src/DocScan/Session/Retrieve/IdDocumentResourceResponse.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Yoti\DocScan\Session\Retrieve;
+
+class IdDocumentResourceResponse extends ResourceResponse
+{
+
+    /**
+     * @var string|null
+     */
+    private $documentType;
+
+    /**
+     * @var string|null
+     */
+    private $issuingCountry;
+
+    /**
+     * @var PageResponse[]
+     */
+    private $pages = [];
+
+    /**
+     * @var DocumentFieldsResponse|null
+     */
+    private $documentFields;
+
+    /**
+     * DocumentResourceResponse constructor.
+     * @param array<string, mixed> $idDocument
+     */
+    public function __construct(array $idDocument)
+    {
+        parent::__construct($idDocument);
+
+        $this->documentType = $idDocument['document_type'] ?? null;
+        $this->issuingCountry = $idDocument['issuing_country'] ?? null;
+
+        if (isset($idDocument['pages'])) {
+            $this->pages = $this->parsePages($idDocument['pages']);
```

Comment:
Could replace private method with:

```suggestion
            foreach ($idDocument['pages'] as $page) {
                $this->pages[] = new PageResponse($page);
            }
```

Reply:
Fixed in 93f3254